src/fetchers/git-fetcher.js

```diff
@@ -51,53 +91,48 @@ export default class GitFetcher extends BaseFetcher {
           if (!expectHash || expectHash === actualHash) {
             resolve({
               hash: actualHash,
-              resolved: `${pathname}#${actualHash}`,
             });
           } else {
             reject(new SecurityError(
-              reporter.lang('fetchBadHash', expectHash, actualHash),
+              this.reporter.lang('fetchBadHash', expectHash, actualHash),
             ));
           }
         })
         .on('error', function(err) {
-          reject(new MessageError(reporter.lang('fetchErrorCorrupt', err.message, localTarball)));
+          reject(new MessageError(this.reporter.lang('fetchErrorCorrupt', err.message, tarballPath)));
         });
     });
   }
 
   async fetchFromExternal(): Promise<FetchedOverride> {
-    const commit = this.hash;
-    invariant(commit, 'Commit hash required');
+    const hash = this.hash; // if we don't store it in a temp variable, flow will not persist its invariant state
```

Comment:
This is quite a common pattern in flow, you probably don't need to add a comment to describe it.
If someone decides to inline the variable flow won't let them anyway :)

Also I think we don't add comments on the same line with code (I might be wrong though)